### PR TITLE
Bag Indexer Skeleton

### PR DESCRIPTION
**Pull request by kenoir**
Simplest possible running implementation of a bag indexer.

For https://github.com/wellcomecollection/platform/issues/4521

### CODE REVIEW

File: indexer/bag_indexer/src/main/resources/application.conf [hunk 1]
```diff
@@ -5,4 +5,4 @@ es.port=${?es_port}
 es.username=${?es_username}
 es.password=${?es_password}
 es.protocol=${?es_protocol}
-es.ingests.index-prefix=${?es_ingests_index_prefix}
+es.bags.index-prefix=${?es_bags_index_prefix}
```
Comment (alexwlchan) [inline indexer/bag_indexer/src/main/resources/application.conf:8]: minor: this is now `index_name`, not `index_prefix`

File: indexer/bag_indexer/src/main/scala/uk/ac/wellcome/platform/archive/indexer/bags/BagsIndexConfig.scala [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import com.sksamuel.elastic4s.ElasticDsl.{dateField, keywordField}
+import com.sksamuel.elastic4s.requests.mappings.FieldDefinition
+import uk.ac.wellcome.platform.archive.indexer.elasticsearch.IndexConfig
+
+object BagsIndexConfig extends IndexConfig {
+  override protected val fields: Seq[FieldDefinition] =
+    Seq(
+      keywordField("id"),
+      dateField("createdDate"),
+      keywordField("type")
```
Comment (alexwlchan) [inline indexer/bag_indexer/src/main/scala/uk/ac/wellcome/platform/archive/indexer/bags/BagsIndexConfig.scala:12]: Do we need to type these objects? We have for the ingests, but that's mostly a holdover from the display models, and I'm not sure it's useful.

File: indexer/bag_indexer/src/main/scala/uk/ac/wellcome/platform/archive/indexer/bags/Main.scala [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    implicit val materializer: Materializer =
+      AkkaBuilder.buildMaterializer()
+
+    implicit val monitoringClient: CloudwatchMetricsMonitoringClient =
+      CloudwatchMonitoringClientBuilder.buildCloudwatchMonitoringClient(config)
+
+    implicit val sqsClient: SqsAsyncClient =
+      SQSBuilder.buildSQSAsyncClient(config)
+
+    val index = Index(name = config.required[String]("es.bags.index-name"))
```
Comment (alexwlchan) [inline indexer/bag_indexer/src/main/scala/uk/ac/wellcome/platform/archive/indexer/bags/Main.scala:44]: This doesn't match the config flag above.

File: indexer/bag_indexer/src/test/scala/uk/ac/wellcome/platform/archive/indexer/bag/BagIndexerFeatureTest.scala [hunk 4]
```diff
@@ -1,10 +1,40 @@
...
+import uk.ac.wellcome.platform.archive.indexer.elasticsearch.Indexer
+
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class BagIndexerFeatureTest
+    extends IndexerFeatureTestCases[StorageManifest, IndexedStorageManifest]
+    with StorageManifestGenerators {
+
+  override def convertIndexedT(t: StorageManifest): IndexedStorageManifest =
+    IndexedStorageManifest(t)
```
Comment (alexwlchan) [inline indexer/bag_indexer/src/test/scala/uk/ac/wellcome/platform/archive/indexer/bag/BagIndexerFeatureTest.scala:23]:
```suggestion
  override def convertIndexedT(manifest: StorageManifest): IndexedStorageManifest =
    IndexedStorageManifest(manifest)
```

File: indexer/bag_indexer/src/test/scala/uk/ac/wellcome/platform/archive/indexer/bag/BagIndexerWorkerTest.scala [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  override def createIndexer(
+    index: Index
+  ): Indexer[StorageManifest, IndexedStorageManifest] =
+    new BagIndexer(
+      client = elasticClient,
+      index = index
+    )
+
+  override def convertToIndexed(t: StorageManifest): IndexedStorageManifest =
+    IndexedStorageManifest(t)
```
Comment (alexwlchan) [inline indexer/bag_indexer/src/test/scala/uk/ac/wellcome/platform/archive/indexer/bag/BagIndexerWorkerTest.scala:39]:
```suggestion
  override def convertToIndexed(manifest: StorageManifest): IndexedStorageManifest =
    IndexedStorageManifest(manifest)
```

File: indexer/ingests_indexer/src/test/scala/uk/ac/wellcome/platform/archive/indexer/ingests/IngestsIndexerFeatureTest.scala [hunk 6]
```diff
@@ -1,57 +1,33 @@
...
 
 class IngestsIndexerFeatureTest
-    extends AnyFunSpec
-    with Matchers
-    with EitherValues
-    with IndexerFixtures[Ingest, IndexedIngest]
+    extends IndexerFeatureTestCases[Ingest, IndexedIngest]
     with IngestGenerators {
 
+  override def convertIndexedT(t: Ingest): IndexedIngest = IndexedIngest(t)
```
Comment (alexwlchan) [inline indexer/ingests_indexer/src/test/scala/uk/ac/wellcome/platform/archive/indexer/ingests/IngestsIndexerFeatureTest.scala:18]:
```suggestion
  override def convertIndexedT(ingest: Ingest): IndexedIngest = IndexedIngest(ingest)
```

File: indexer/bag_indexer/src/main/scala/uk/ac/wellcome/platform/archive/indexer/bags/BagsIndexConfig.scala (hunk 2 above)
Comment (kenoir) [reply indexer/bag_indexer/src/main/scala/uk/ac/wellcome/platform/archive/indexer/bags/BagsIndexConfig.scala:12]: I'm not sure - i'll think about that more thoroughly when we apply the proper mappings.

**Pull request merged**